src/components/radio/daikin-radio.ts

```diff
@@ -0,0 +1,226 @@
+ */
+@customElement("daikin-radio")
+class DaikinRadio extends LitElement implements DaikinRadioProps {
+  static styles = css`
+    ${unsafeCSS(tailwindStyles)}
+
+    :host {
+      display: inline-block;
+    }
+  `;
+
+  private _handleClick(event: MouseEvent) {
+    if (this.readonly) {
+      event.preventDefault();
+    }
+  }
+
+  static formAssociated = true;
+
+  // define internals to let radio can be used in form
+  @property({ type: Object })
+  internals;
+
+  constructor() {
+    super();
+    this.internals = this.attachInternals();
+  }
+
+  private _updateFormValue() {
+    if (this.checked) {
+      this.internals.setFormValue(this.value);
+    } else {
+      this.internals.setFormValue(null);
+    }
+  }
+
+  updated(changedProperties: Map<string, any>) {
+    if (changedProperties.has("checked")) {
+      this._updateFormValue();
+    }
+  }
+
+  private _handleChange(event: Event) {
+    const input = this.shadowRoot?.querySelector("input") as HTMLInputElement;
+    this.checked = input.checked;
+    this._updateFormValue();
+    const newEvent = new Event("change", event);
+    this.dispatchEvent(newEvent);
+  }
+
+  /**
+   * Specify the label text for check box
+   */
+  @property({ type: String })
+  label = "";
+
+  /**
+   * Specify the component size
+   */
+  @property({ type: String })
+  size: ComponentSizeProps = "small";
+
+  /**
+   * Specify the label position
+   * when `left` the label will be in left of radio, when `right` label will be in right of radio
+   */
+  @property({ type: String, attribute: "label-position" })
+  labelPosition: "left" | "right" = "right";
+
+  /**
+   * Specify whether the Radio should be disabled
+   */
+  @property({ type: Boolean, reflect: true })
+  disabled = false;
+
+  /**
+   * Specify whether the radio is read only
+   */
+  @property({ type: Boolean, reflect: true })
+  readonly = false;
+
+  /**
+   * Specify whether the radio is be checked
+   */
+  @property({ type: Boolean, reflect: true })
+  checked = false;
+
+  /**
+   * The form name.
+   */
+  @property({ type: String, reflect: true })
+  name = "";
+
+  /**
+   * The value.
+   */
+  @property({ type: String, reflect: true })
+  value = "";
+
+  /**
+   * Specify whether the Radio is in a error state
+   */
+  @property({ type: Boolean, reflect: true })
+  error = false;
+
+  render() {
+    // Specify the component size
+    const labelClassName = labelCN({ size: this.size });
+    const radioClassName = radioCN({ size: this.size });
+
+    const labelText = this.label
+      ? html`<span class="${labelClassName}">${this.label}</span>`
+      : html``;
+    const inputTag = html`<input
+        class="${hideRadioCN}"
+        type="radio"
+        name="${this.name}"
+        value="${this.value}"
+        ?checked=${this.checked}
+        ?readonly=${this.readonly}
+        ?disabled=${this.disabled}
+        @click=${this._handleClick}
+        @change=${this._handleChange}
```

Comment:
This might lose `this`.

```suggestion
        @click=${this._handleClick.bind(this)}
        @change=${this._handleChange.bind(this)}
```